pipelines/azure_pipelines_nightly.yml

```diff
@@ -0,0 +1,211 @@
+trigger:
+  branches:
+    include:
+    - master
+resources:
+  repositories:
+  - repository: self
+    type: git
+    ref: master
+pr:
+  - master
+
+variables:
+  BuildPlatform: 'AnyCpu'
+  BuildConfiguration: 'Release'
+  RootDir: '$(Build.SourcesDirectory)'
+  ProductBinPath: '$(rootDir)\bin\$(BuildPlatform)\$(BuildConfiguration)'
+  RELEASE_SEMANTICS: '$(ReleaseSemantics)'
+  SourcesRoot: '$(RootDir)'
+
+jobs:
+  - job: Main
+    displayName: Main Build
+    # setting a 3hour timeout as webapi tests normally take about 2hr 30 mins
```

Comment:
What if more tests are added and they exceed the 3hr limit?

Reply:
Then that would need us to update this variable. So far not sure if this would happen anytime soon.